tools/dev/tests/test_poetrypkg.py

```diff
@@ -0,0 +1,60 @@
+import subprocess
+import sys
+
+import pytest
+
+from pathlib import Path
+
+from click.testing import CliRunner
+
+cli = Path(__file__).parent.parent / "poetrypkg.py"
+with open(cli, encoding='utf-8') as f:
+    exec(f.read())
+
+
+@pytest.mark.skipif(sys.version_info >= (3, 8), reason="Developer Python minimum is 3.8")
```

Comment:
We might want to invert this to skip if < 3.8. I feel like this should work on 3.7+, but agreed that the extra dep on backports.cached_property seems more tedious than useful purely for release testing.